From e288aa117377e75dc2fde606cbc39649b9957dce Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Tue, 4 Aug 2015 16:08:51 +0800
Subject: [PATCH 0047/5242] MLK-11318-4 ARM: dts: imx7d-12x12-lpddr3-arm2: add
 dts support for pwm backlight

commit  d640d7cbb0a1fb3cc226c9660462d9fde6840596 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add dts support for pwm backlight on imx_4.1.y branch.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 1793e13..e94a10c 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -14,6 +14,14 @@
 	model = "Freescale i.MX7 LPDDR3 12x12 ARM2 Board";
 	compatible = "fsl,imx7d-12x12-lpddr3-arm2", "fsl,imx7d";
 
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm1 0 5000000>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+		status = "okay";
+	};
+
 	regulators {
 		compatible = "simple-bus";
 
@@ -319,6 +327,12 @@
 			>;
 		};
 
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX7D_PAD_GPIO1_IO08__PWM1_OUT 0x110b0
+			>;
+		};
+
 		pinctrl_uart1_1: uart1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
@@ -419,6 +433,12 @@
 	};
 };
 
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
+
 &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1_1>;
-- 
1.7.9.5

